apps/web/src/routes/(home)/components/CtaBlock.svelte

```diff
@@ -45,6 +46,33 @@
 	function handleClickOutside() {
 		showSelect = false;
 	}
+
+	async function detectDefaultDownload() {
+		const platform = navigator.userAgent.toLowerCase();
+
+		if (platform.includes('mac')) {
+			const userAgentData = await navigator.userAgentData?.getHighEntropyValues(['architecture']);
+			if (userAgentData?.architecture === 'arm' || userAgentData === undefined) {
+				return jsonLinks.downloads.appleSilicon;
+			} else {
+				return jsonLinks.downloads.intelMac;
+			}
+		}
+
+		if (platform.includes('win')) {
+			return jsonLinks.downloads.windowsMsi;
+		}
+
+		if (platform.includes('linux')) {
+			return jsonLinks.downloads.linuxAppimage;
+		}
+	}
+
+	onMount(() => {
+		detectDefaultDownload().then(
+			(platformDownload) => platformDownload && targetDownload.set(platformDownload)
+		);
+	});
```

Comment:
We should put the logic to determine the default inside `store.ts` so it doesn't get lost if we ever replace this component.